pkg/build/webhook/generic/generic.go

```diff
@@ -94,6 +99,35 @@ func (p *WebHookPlugin) Extract(buildCfg *api.BuildConfig, secret, path string,
 	return revision, true, nil
 }
 
+func updateBuildEnv(buildCfg *api.BuildConfig, env []kapi.EnvVar) {
+	var buildEnv *[]kapi.EnvVar
+
+	switch {
+	case buildCfg.Spec.BuildSpec.Strategy.SourceStrategy != nil:
+		buildEnv = &buildCfg.Spec.BuildSpec.Strategy.SourceStrategy.Env
+	case buildCfg.Spec.BuildSpec.Strategy.DockerStrategy != nil:
+		buildEnv = &buildCfg.Spec.BuildSpec.Strategy.DockerStrategy.Env
+	case buildCfg.Spec.BuildSpec.Strategy.CustomStrategy != nil:
+		buildEnv = &buildCfg.Spec.BuildSpec.Strategy.CustomStrategy.Env
+	}
+
+	newEnv := []kapi.EnvVar{}
+	for _, e := range *buildEnv {
+		exists := false
+		for _, n := range env {
+			if e.Name == n.Name {
+				exists = true
+				break
+			}
+		}
+		if !exists {
+			newEnv = append(newEnv, e)
+		}
+	}
+	newEnv = append(newEnv, env...)
+	*buildEnv = newEnv
+}
```

Comment:
Since this appears to be an exact copy of the updateBuildEnv in generator.go, it would be better to just make that function public and call it rather than making a copy of it.